### PR TITLE
Maize in season + random forest

### PR DESCRIPTION
**Pull request by gabrieltseng**
Random forest results:

| Class | PA | UA | F1 |
| --- | --- | --- | --- |
| Water | 0.799 | 0.787 | 0.793 |
| Bare Ground | 0.787 | 0.410 | 0.539 | 
| Rangeland | 0.353 | 0.049 | 0.086 |
| Flooded Vegetation | 0.446 | 0.245 | 0.316 |
| Trees | 0.672 | 0.786 | 0.724 |
| Cropland | 0.507 | 0.860 | 0.638 |
| Buildings | 0.755 | 0.697 | 0.725 |
| Overall Accuracy | | | 0.611 | 

| Class | PA | UA | F1 |
| --- | --- | --- | --- |
| Corn | 0.886 | 0.758 | 0.817 |
| Cassava | 0.267 | 0.328 | 0.294 |
| Rice | 0.924 | 0.926 | 0.926 |
| Sesame | 0.316 | 0.031 | 0.057 |
| Beans | 0.251 | 0.811 | 0.383 |
| Millet | 0.000 | 0.000 | 0.000 |
| Sorghum | 0.000 | 0.000 | 0.000 |
| Overall Accuracy | | | 0.710 |

### CODE REVIEW

File: olmoearth_projects/projects/mozambique_lulc/train_random_forest.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+                            "B06",
+                            "B07",
+                            "B8A",
+                            "B11",
+                            "B12",
+                            "B01",
+                            "B09",
+                        ]
+                    }
+                )
```
Comment (yawenzzzz) [inline olmoearth_projects/projects/mozambique_lulc/train_random_forest.py]: Suggestion: RandomForest is flexible about feature value ranges, so maybe dropping the OlmoEarth norm statistics (or just apply standard min/max value to normalize the bands) and see what's the overall accuracy.

File: olmoearth_projects/projects/mozambique_lulc/train_random_forest.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+def train_random_forest(
+    data: AllData, crop_type: bool, include_val_in_train: bool = False
+) -> dict[str, float]:
+    """Train a random forest and record metrics."""
+    if include_val_in_train:
+        x = np.concatenate([data.x_train, data.x_val], axis=0)
+        y = np.concatenate([data.y_train, data.y_val], axis=0)
+    else:
+        x, y = data.x_train, data.y_train
+    model = RandomForestClassifier().fit(x, y)
```
Comment (yawenzzzz) [inline olmoearth_projects/projects/mozambique_lulc/train_random_forest.py]: Do we have their RF parameters? like `n_estimators` and `max_depth`, not sure if the defaults are good enough or not. Also, will be helpful to add `random_state` to make the single run reproducible.

File: olmoearth_projects/projects/mozambique_lulc/train_random_forest.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+from rslearn.models.olmoearth_pretrain.norm import OlmoEarthNormalize
+from rslearn.train.dataset import DataInput, ModelDataset, SplitConfig
+from rslearn.train.tasks.multi_task import MultiTask
+from rslearn.train.tasks.segmentation import SegmentationTask
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import accuracy_score
+from tqdm import tqdm
+from upath import UPath
+
+CLASS_MAP = {
```
[the remark this answers is not in the document]
Comment (gabrieltseng) [reply olmoearth_projects/projects/mozambique_lulc/train_random_forest.py]: updated to LULC_CLASS_MAP
